controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1422,6 +1436,11 @@ func (r *BareMetalHostReconciler) doServiceIfNeeded(prov provisioner.Provisioner
 	}
 	if provResult.ErrorMessage != "" {
 		info.host.Status.Provisioning.Firmware = nil
+		if hfcDirty {
+			hfc.Status.Updates = nil
+			// Probably safe to ignore any errors, since we will return result based on an error
```

Comment:
I don't quite get this comment. It's not safe to ignore errors: this way we will not retry Status.Update next time, and `Status.Updates` will keep whatever value is has. I think what @zaneb meant was that if we return actionError here, on the next iteration the provisioner.Service call will still return an ErrorMessage, so we'll enter this code again.

On top of that, we should not call Status.Update if the value is already `nil`.

Reply:
Done, thanks!